test-app/runtime/src/main/cpp/v8_inspector/src/inspector/v8-overlay-agent-impl.cpp

```diff
@@ -0,0 +1,79 @@
+//
+// Created by pkanev on 11/15/2017.
+//
+
+#include <NativeScriptAssert.h>
+#include <ArgConverter.h>
+#include <v8_inspector/src/inspector/utils/v8-inspector-common.h>
+#include "v8-overlay-agent-impl.h"
+
+namespace v8_inspector {
+
+using tns::ArgConverter;
+
+namespace OverlayAgentState {
+static const char overlayEnabled[] = "overlayEnabled";
+}
+
+V8OverlayAgentImpl::V8OverlayAgentImpl(V8InspectorSessionImpl* session, protocol::FrontendChannel* frontendChannel,
+                                       protocol::DictionaryValue* state)
+    : m_session(session),
+      m_frontend(frontendChannel),
+      m_state(state),
+      m_enabled(false) {
+}
+
+V8OverlayAgentImpl::~V8OverlayAgentImpl() { }
+
+void V8OverlayAgentImpl::enable(ErrorString* errorString) {
+    if (m_enabled) {
+        *errorString = "Overlay Agent already enabled!";
+        return;
+    }
+
+    m_state->setBoolean(OverlayAgentState::overlayEnabled, true);
+    m_enabled = true;
+}
+
+void V8OverlayAgentImpl::disable(ErrorString*) {
+    if (!m_enabled) {
+        return;
+    }
+
+    m_state->setBoolean(OverlayAgentState::overlayEnabled, false);
+
+    m_enabled = false;
+}
+
+void V8OverlayAgentImpl::setShowFPSCounter(ErrorString*, bool in_show) {
```

Comment:
maybe do something like: 
```
*errorString = "<method name> Not implemented ";
```
just in case 
for this and the other empty methods.

Reply:
I'll add the errorStrings, you are right about those. As for the generated files - there's a remark emitted by the generator that the files in question are generated, have a look at https://github.com/NativeScript/android-runtime/blob/1541eb17aa6784a75e88d6093e8fa59000d43d74/test-app/runtime/src/main/cpp/v8_inspector/src/inspector/protocol/CSS.cpp#L1

Reply:
missed the comment, my bad